native/src/main/java/io/ballerina/stdlib/mime/util/EntityBodyHandler.java

```diff
@@ -435,7 +447,18 @@ public void notifySuccess(Object result) {
                     EntityBodyHandler.closeMessageOutputStream(outputStream);
                     return;
                 }
-                writeContentPart((BMap) result, outputStream);
+                if (result instanceof BError error) {
+                    entity.addNativeData(ENTITY_BYTE_STREAM, null);
```

Comment:
Can't we call the `notifyFailure` here also?

Reply:
fixed with: [b4a3de6](https://github.com/ballerina-platform/module-ballerina-mime/pull/530/commits/b4a3de60a9fb06991575b155f3ed1d34d0121599)